examples/gno.land/p/demo/memeland/memeland.gno

```diff
@@ -0,0 +1,171 @@
+package memeland
+
+import (
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Post - as previously defined
+type Post struct {
+	ID            string
+	Data          string
+	Author        std.Address
+	Timestamp     time.Time
+	UpvoteTracker *avl.Tree // address > struct{}{}
+}
+
+type Memeland struct {
+	Posts       *avl.Tree
+	MemeCounter seqid.ID
+}
+
+func NewMemeland() *Memeland {
+	return &Memeland{
+		Posts: avl.NewTree(),
+	}
+}
+
+// PostMeme - Adds a new post
+func (m *Memeland) PostMeme(data string, timestamp int64) string {
+	if data == "" || timestamp <= 0 {
+		return "timestamp or data cannot be empty"
+	}
+
+	// Generate ID
+	id := m.MemeCounter.Next().String()
+
+	newPost := &Post{
+		ID:            id,
+		Data:          data,
+		Author:        std.GetOrigCaller(),
+		Timestamp:     time.Unix(timestamp, 0),
+		UpvoteTracker: avl.NewTree(),
+	}
+
+	m.Posts.Set(id, newPost)
+
+	return id
+}
+
+func (m *Memeland) Upvote(id string) string {
```

Comment:
Please make sure this method only returns a string in the case of a successful outcome, and panics otherwise

This pattern should be applied everywhere

Reply:
[0ceba4c](https://github.com/gnolang/gno/pull/1751/commits/0ceba4cb79482d5f05961f81f869d92bad405b63)